swarm/storage/mru/update.go

```diff
@@ -17,304 +17,128 @@
 package mru
 
 import (
-	"bytes"
 	"encoding/binary"
 	"errors"
-	"fmt"
-	"hash"
 
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/swarm/log"
 	"github.com/ethereum/go-ethereum/swarm/multihash"
 	"github.com/ethereum/go-ethereum/swarm/storage"
 )
 
-// Signature is an alias for a static byte array with the size of a signature
-const signatureLength = 65
-
-type Signature [signatureLength]byte
-
-// SignedResourceUpdate contains signature information about a resource update
-type SignedResourceUpdate struct {
-	resourceUpdate // actual content that will be put on the chunk, less signature
-	signature      *Signature
-	updateAddr     storage.Address // resulting chunk address for the update
```

Comment:
`SignedResourceUpdate` was moved to its own file, leaving this file mostly empty.
However, I added new code, thus this diff looks horrible.
The real diff is here:
http://www.mergely.com/OvoJels7/
This is the last one of this clusterf**cks. I apologize.
`UpdateHeader` moved to its own file.